awx/main/fields.py

```diff
@@ -740,8 +739,24 @@ def validate(self, value, model_instance):
 
         class TowerNamespace:
             filename = None
-
         valid_namespace['tower'] = TowerNamespace()
+
+        # ensure either single file or multi-file syntax is used (but not both)
+        template_names = set(key for type_, injector in value.items()
```

Comment:
This double-nested loop is a little hairy to read.  You're also needlessly looping over injectors (like `env`) that don't actually _support_ the `template` key.  Can you just look at the `file` key instead?

Reply:
Will give that a try - now that I look at the double for loop again, I can't even make out what I was doing. Hard to read is right.